common/src/main/java/com/microsoft/identity/common/migration/DefaultSharedPrefsFileManagerReencrypter.java

```diff
@@ -0,0 +1,188 @@
+        final String methodName = ":reencrypt (sync)";
+        final Map<String, String> cacheEntries = new HashMap<>(fileManager.getAll());
+        Logger.verbose(TAG + methodName,
+                "Attempting to migrate cache entries: " + cacheEntries.size());
+        final MigrationOperationResult result = new MigrationOperationResult();
+        result.setTotalRecords(cacheEntries.size());
+        final Set<String> keysMarkedForRemoval = new HashSet<>();
+        final AtomicBoolean shouldAbort = new AtomicBoolean(false);
+        final Set<String> skipKeys = new HashSet<>();
+
+        // Decrypt everything
+        applyCacheMutation(
+                cacheEntries,
+                new Callable<Map.Entry<String, String>>() {
+                    @Override
+                    public void call(final Map.Entry<String, String> entry) throws Exception {
+                        final String decryptedText = decrypter.decrypt(entry.getValue());
+                        entry.setValue(decryptedText);
+                    }
+                },
+                result,
+                params,
+                keysMarkedForRemoval,
+                skipKeys,
+                shouldAbort
+        );
+
+        // Clear any keys marked for removal...
+        clearEntriesMarkedForRemoval(fileManager, cacheEntries, keysMarkedForRemoval);
+
+        if (shouldAbort.get()) {
+            Logger.info(TAG + methodName, "Aborting after decrypt.");
+            return result;
+        }
+
+        // Reencrypt everything
+        applyCacheMutation(
+                cacheEntries,
+                new Callable<Map.Entry<String, String>>() {
+                    @Override
+                    public void call(final Map.Entry<String, String> entry) throws Exception {
+                        final String reencryptedText = encrypter.encrypt(entry.getValue());
+                        entry.setValue(reencryptedText);
+                    }
+                },
+                result,
+                params,
+                keysMarkedForRemoval,
+                skipKeys,
+                shouldAbort
+        );
+
+        // Clear any keys marked for removal...
+        clearEntriesMarkedForRemoval(fileManager, cacheEntries, keysMarkedForRemoval);
+
+        if (shouldAbort.get()) {
+            Logger.info(TAG + methodName, "Aborting after reencrypt.");
+            return result;
+        }
+
+        // Write the newly encrypted records...
+        Logger.info(TAG + methodName, "Writing reencrypted cache entries.");
+        for (final Map.Entry<String, String> cacheEntry : cacheEntries.entrySet()) {
+            fileManager.putString(cacheEntry.getKey(), cacheEntry.getValue());
+        }
+
+        return result;
+    }
+
+    private interface Callable<T> {
+        void call(T t) throws Exception;
+    }
+
+    private void applyCacheMutation(@NonNull final Map<String, String> cacheEntries,
```

Comment:
Even though this is private, I think javadoc would be nice, primarily is because there are a lot of params and it seems to be doing a lot of things

Reply:
ACK

Reply:
To keep this work easy to `cherry-pick` for @AdamBJohnsonx - I have applied your feedback in a single amended commit via `32fdc9cdf4a9357bd40cca2e0e08f6daf1d50157`